### PR TITLE
[MM-54529] Prevent Persistent Notifications Error Message for DM Channel

### PR DESCRIPTION
**Pull request by escofresco**
#### Summary
Added condition so 'Recipients must be @mentioned' no longer appears for messages sent in DM channel when **Apply**ing **Send persistent notifications** was chosen by selecting the **Message Priority** on [Enterprise and Professional plans](https://docs.mattermost.com/collaborate/message-priority.html)  like below:
   

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-mobile/issues/25116
Jira https://mattermost.atlassian.net/browse/MM-54529

#### Checklist
- [x] Includes text changes and localization file updates
- [x] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: Android emulator

before | after
-|-
Screenshot 2023-11-21 at 5 34 58 AM|Screenshot 2023-11-21 at 5 35 37 AM

#### Release Note
```release-note
Added channelType param to persistentNotificationsConfirmation.
```

### CODE REVIEW

File: app/utils/post/index.ts [hunk 1]
```diff
@@ -152,7 +152,7 @@ export async function persistentNotificationsConfirmation(serverUrl: string, val
             });
             description = intl.formatMessage({
                 id: 'persistent_notifications.confirm.description',
-                defaultMessage: '@mentioned recipients will be notified every {interval, plural, one {minute} other {{interval} minutes}} until they’ve acknowledged or replied to the message.',
+                defaultMessage: `${ 'R' ? channelType === General.DM_CHANNEL : '@mentioned r' }ecipients will be notified every {interval, plural, one {minute} other {{interval} minutes}} until they’ve acknowledged or replied to the message.`,
```
Comment (enahum) [inline app/utils/post/index.ts:155]: we cannot have if statements in translatable strings...

File: app/components/post_draft/draft_input/index.tsx [hunk 2]
```diff
@@ -147,28 +147,28 @@ export default function DraftInput({
...
     const handleSendMessage = useCallback(async () => {
-        if (persistenNotificationsEnabled) {
-            persistentNotificationsConfirmation(serverUrl, value, mentionsList, intl, sendMessage, persistentNotificationMaxRecipients, persistentNotificationInterval);
+        if (persistentNotificationsEnabled) {
+            persistentNotificationsConfirmation(serverUrl, value, mentionsList, intl, sendMessage, persistentNotificationMaxRecipients, persistentNotificationInterval, channelType);
         } else {
             sendMessage();
         }
-    }, [serverUrl, mentionsList, persistenNotificationsEnabled, persistentNotificationMaxRecipients, sendMessage, value]);
+    }, [serverUrl, mentionsList, persistentNotificationsEnabled, persistentNotificationMaxRecipients, sendMessage, value]);
```
Comment (enahum) [inline app/components/post_draft/draft_input/index.tsx:171]: `channelType` seems to be missing in the dependencies, I don't expect it to change in the case of a DM, but anyway, missing.

**Review by enahum — CHANGES_REQUESTED** [review]

**Review by larkox — COMMENTED** [review]
Apart of the comments already made, LGTM.

File: app/utils/post/index.ts (hunk 1 above)
Comment (escofresco) [reply app/utils/post/index.ts:155]: Oh true, fixing now.

**Comment by escofresco** [comment]
Need help with testing in app: In `config.json` on server side, "Allow persistent notifications" is set to true. Still not able to actually see the feature. Guessing because it's for pro/enterprise. Could someone point me to info on how to enable it for testing?

**Review by harshilsharma63 — APPROVED** [review]
👌👌👌

**Comment by enahum** [comment]
> > Need help with testing in app: In `config.json` on server side, "Allow persistent notifications" is set to true. Still not able to actually see the feature. Guessing because it's for pro/enterprise. Could someone point me to info on how to enable it for testing?
> 
> 
> 
> Are you trying to test this on community.mattermost.com ? I think the best option will be to create a test cloud server with persistent notifications enabled. @enahum do you know how to do that?

Sorry, I never use cloud servers for testing only my local instance.. maybe we can ask someone from QA?

**Comment by hmhealey** [comment]
We can set up test servers using `/cloud` on Community. I don't know if it has an option to add a license, but I added the dev one to it.

---

@escofresco If you want to use https://harrison-test.test.mattermost.cloud for it, I set that up a week or two ago and added a license to it. You should be free to create a team and any number of users on it, and it has persistent notifications enabled, so you can use it for testing this PR

**Comment by larkox** [comment]
@escofresco Are you still looking into this?

**Comment by escofresco** [comment]
@larkox Yes, forgot about it but I will test the change right now

**Comment by escofresco** [comment]
@hmhealey thanks for the test server, I got a chance to test the change. @larkox all good to go

**Review by enahum — APPROVED** [review]

**Review by yasserfaraazkhan — CHANGES_REQUESTED** [review]
@larkox on server i had set 11 mins time for notification, but in mobile its showing 0.
would it be related to the PR change? 

image
rn_image_picker_lib_temp_8703a033-68a3-4135-8a68-38aeb4c4adab

**Comment by larkox** [comment]
@yasserfaraazkhan Looking at the code, it shouldn't interfere, but I am not 100% sure (it touches parts that are near that logic). Could you double check on main, and if it happens there, create a new ticket?

**Comment by enahum** [comment]
> Not sure why 'build-ios-pr' is failing but I'm only able to build to test android since i don't have apple developer account

re-triggered the job, build was a success.

@mvitale1989 FYI

**Pull request merged**